src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CitizenCoreCaseDataService.java

```diff
@@ -110,92 +120,35 @@ private EventRequestData eventRequest(CaseEvent caseEvent, String userId) {
-                eventRequestData.getUserId(),
-                eventRequestData.getJurisdictionId(),
-                eventRequestData.getCaseTypeId(),
-                caseId.toString(),
-                eventRequestData.getEventId()
-            );
-        }
-    }
-
-    private CaseDetails submitUpdate(
-        String authorisation,
-        EventRequestData eventRequestData,
-        CaseDataContent caseDataContent,
-        Long caseId,
-        boolean isRepresented
-    ) {
-        if (isRepresented) {
-            return coreCaseDataApi.submitEventForCaseWorker(
-                authorisation,
-                authTokenGenerator.generate(),
-                eventRequestData.getUserId(),
-                eventRequestData.getJurisdictionId(),
-                eventRequestData.getCaseTypeId(),
-                caseId.toString(),
-                eventRequestData.isIgnoreWarning(),
-                caseDataContent
-            );
-        } else {
-            return coreCaseDataApi.submitEventForCitizen(
-                authorisation,
-                authTokenGenerator.generate(),
-                eventRequestData.getUserId(),
-                eventRequestData.getJurisdictionId(),
-                eventRequestData.getCaseTypeId(),
-                caseId.toString(),
-                eventRequestData.isIgnoreWarning(),
-                caseDataContent
-            );
-        }
-    }
-
     public CaseDetails updateCase(
         String authorisation,
         Long caseId,
-        CaseData caseData,
         CaseEvent caseEvent
     ) {
         try {
             UserDetails userDetails = idamClient.getUserDetails(authorisation);
-            EventRequestData eventRequestData = eventRequest(caseEvent, userDetails.getId());
+            EventRequestData eventRequestData = coreCaseDataService.eventRequest(caseEvent, userDetails.getId());
 
-            StartEventResponse startEventResponse = startUpdate(
-                authorisation,
-                eventRequestData,
-                caseId,
-                !userDetails.getRoles().contains(CITIZEN_ROLE)
+            StartEventResponse startEventResponse =
+                coreCaseDataService.startUpdate(
+                    authorisation,
+                    eventRequestData,
+                    String.valueOf(caseId),
+                    true
+                );
+
+            CaseData caseDataUpdated = CaseUtils.getCaseDataFromStartUpdateEventResponse(startEventResponse, objectMapper);
+            CaseDataContent caseDataContent = coreCaseDataService.createCaseDataContent(
+                startEventResponse,
+                caseDataUpdated
             );
-            Map<String, Object> caseDataMap = caseData.toMap(objectMapper);
-            Iterables.removeIf(caseDataMap.values(), Objects::isNull);
-            CaseDataContent caseDataContent = caseDataContent(startEventResponse, caseDataMap);
-            return submitUpdate(
```

Comment:
@remya-james , what happened to removed codes

Reply:
@AlokDatta This is replaced with the case data retrieved from CaseUtils.getCaseDataFromStartUpdateEventResponse(startEventResponse, objectMapper);
Lines 76-79